presto-spark-base/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeExecution.java

```diff
@@ -42,4 +64,33 @@ public void testNativeExecutionWithTableWrite()
         // Expecting 0 row updated since currently the NativeExecutionOperator is dummy.
         assertUpdate(session, "CREATE TABLE test_tablescan as SELECT orderkey, custkey FROM orders", 0);
     }
+
+    @Test(priority = 2, dependsOnMethods = "testNativeExecutionWithTableWrite")
+    public void testNativeExecutionShuffleManager()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(NATIVE_EXECUTION_ENABLED, "true")
+                .setSystemProperty("table_writer_merge_operator_enabled", "false")
+                .setCatalogSessionProperty("hive", "collect_column_statistics_on_write", "false")
+                .build();
+
+        PrestoSparkQueryRunner queryRunner = (PrestoSparkQueryRunner) getQueryRunner();
+
+        // Reset the spark context to register the native execution shuffle manager. We want to let the query runner use the default spark shuffle
+        // manager to generate the test tables and only test the new native execution shuffle manager on the test below test cases.
+        queryRunner.resetSparkContext(getNativeExecutionShuffleConfigs());
+        // Expecting 0 row updated since currently the NativeExecutionOperator is dummy.
+        queryRunner.execute(session, "CREATE TABLE test_aggregate as SELECT  partkey, count(*) c FROM lineitem WHERE partkey % 10 = 1 GROUP BY partkey");
+
+        assertNotNull(SparkEnv.get());
+        assertTrue(SparkEnv.get().shuffleManager() instanceof PrestoSparkNativeExecutionShuffleManager);
+        PrestoSparkNativeExecutionShuffleManager shuffleManager = (PrestoSparkNativeExecutionShuffleManager) SparkEnv.get().shuffleManager();
+        Optional<ShuffleHandle> shuffleHandle = shuffleManager.getShuffleHandle(0);
+        assertTrue(shuffleHandle.isPresent());
+        assertTrue(shuffleHandle.get() instanceof BypassMergeSortShuffleHandle);
+        BypassMergeSortShuffleHandle bypassMergeSortShuffleHandle = (BypassMergeSortShuffleHandle) shuffleHandle.get();
```

Comment:
Can use specialize the type for `BypassMergeSortShuffleHandle`: `BypassMergeSortShuffleHandle<.., ..>`